File: C#/KataClasses/CardMask.cs
using System;
using System.Collections.Generic;

// Codewars

namespace Katas.KataClasses;

public static class CardMask
{
    public static string Maskify(string cc)
    {
        char[] ccArray = cc.ToCharArray();
        List<String> maskedList = new List<String> { };

        for (int i = 0; i < ccArray.Length; i++)
        {
            if (i < (ccArray.Length - 4))
                maskedList.Add("#");
            else
                maskedList.Add(ccArray[i].ToString());
        }

        return string.Join("", maskedList);

        // Using Regex.Replace(cc, {REGEX HERE}, "#") would have worked too.
    }
}

File: TS_JS/TypeScript/KataClasses/cardMask.ts
// Codewars

export function maskify(cc: string): string
{
    let ccArray = cc.split('');
    let maskedList: string[] = [];

    for (let i = 0; i < ccArray.length; i++) {
        if (i < (ccArray.length - 4)) {
            maskedList.push("#");
        } else {
            maskedList.push(ccArray[i].toString());
        }
    }

    return maskedList.join("");
}

export function maskify2ElectricBoogaloo(cc: string): string
{
    return cc.replace(/.(?=.{4})/g, "#");
}